language/move-core/types/src/value.rs

```diff
@@ -1,17 +1,27 @@
 // Copyright (c) The Diem Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::account_address::AccountAddress;
+use crate::{account_address::AccountAddress, identifier::Identifier};
 use anyhow::Result as AResult;
+use serde::ser::SerializeMap;
 use serde::{
     de::Error as DeError,
     ser::{SerializeSeq, SerializeTuple},
     Deserialize, Serialize,
 };
-use std::fmt::{self, Debug};
+use std::marker::PhantomData;
+use std::{
+    collections::BTreeMap,
+    fmt::{self, Debug},
+};
 
 #[derive(Debug, PartialEq, Eq, Clone)]
-pub struct MoveStruct(Vec<MoveValue>);
+pub enum MoveStruct {
+    /// The representation used by the MoveVM
+    Runtime(Vec<MoveValue>),
```

Comment:
MoveVM won't use this struct in the runtime so I think getting field name by default makes sense here?

Reply:
It is currently used in runtime.rs (or at least, `MoveValue` is). Maybe it doesn't need to be used there, but my goal is not cleaning up existing uses; rather, I want to generalize this to allow both representations.